addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -143,6 +143,29 @@ def _timesheet_prepare_line_values(self, index, employee_id, work_hours_data, da
             'company_id': employee_id.company_id.id,
         }
 
+    def _generate_public_time_off_timesheets(self, employees):
+        timesheet_vals_list = []
+        work_hours_data = self._work_time_per_day()
+        for leave in self:
+            for employee in employees:
+                if employee.resource_calendar_id != leave.calendar_id:
+                    continue
+                work_hours_list = work_hours_data[leave.id]
+                for index, (day_date, work_hours_count) in enumerate(work_hours_list):
+                    existing_timesheets = leave["timesheet_ids"].filtered(
```

Comment:
Equivalent to: 
```suggestion
                    existing_timesheets = leave.timesheet_ids.filtered(
```

since leave is a recordset.